webapp/src/ts/modules/contacts/contacts-edit.component.ts

```diff
@@ -235,24 +232,17 @@ export class ContactsEditComponent implements OnInit, OnDestroy, AfterViewInit {
     }
   }
 
-  private async getTitleKey(contact) {
-    const typeId = this.contactTypesService.getTypeId(contact) || this.routeSnapshot.params?.type;
-    const type = await this.contactTypesService.get(typeId);
-    if (!type) {
-      console.error(`Unknown contact type "${typeId}"`);
-      return;
-    }
-
-    return contact ? type.edit_key : type.create_key;
+  private getTitleKey(contact, contactType) {
+    return contact ? contactType.edit_key : contactType.create_key;
```

Comment:
This seems trivial now. If this is not called multiple times, we should just inline the assignment and condition. It's also duplicated with the code from getForm.

Reply:
Good call. I wouldn't deduplicate the code from `getForm` though because there is more than just one assignment in both branches of the if condition

Reply:
I think getting the title should be done in just one place, and use the same key, ideally in the same "place".

Reply:
My first guess would be to store it in a new class property and access it where needed like this `this.titleKey`

Reply:
We're not using class properties for anything else - you added the "context" which we pass from one function to another instead of class properties.

Reply:
So I think we should follow that pattern and somehow include it in the context, ready to be used after it had been calculated once.